src/sphinxvhdl/vhdl.py

```diff
@@ -167,50 +167,78 @@ def run(self):
                 fields = self.get_fields_from_definition(self.content[index])
-                has_groups = has_groups or len(fields) == 4
+                has_groups = has_groups or len(fields) >= 4
+                has_group_desc = has_groups and len(fields) == 5
             index += 1
         index = 0
 
         row: Optional[nodes.row]
         row = nodes.row()
-        if has_groups:
-            row += nodes.entry('', nodes.paragraph('', nodes.Text(self.table_headers[0])))
-            pass
+        # Define first line of table
         row += nodes.entry('', nodes.paragraph('', nodes.Text(self.table_headers[1])))
         row += nodes.entry('', nodes.paragraph('', nodes.Text(self.table_headers[2])))
         row += nodes.entry('', nodes.paragraph('', nodes.Text(self.table_headers[3])))
         row += nodes.entry('', nodes.paragraph('', nodes.Text(self.table_headers[4])))
         head += row
+
         row = None
         current_description_lines = StringList()
         description_entry = nodes.entry('')
-        if has_groups:
-            group += nodes.colspec(colwidth=10)
-            pass
+
+        # Define table fields and their dimensions
         group += nodes.colspec(colwidth=10)
         group += nodes.colspec(colwidth=10)
         group += nodes.colspec(colwidth=10)
-        group += nodes.colspec(colwidth=70)
+        group += nodes.colspec(colwidth=100)
+
+        # Insert group header to table
         group += head
         group += body
+        current_group = ''
+
+        # Fill the table with content
         while len(self.content) > index:
             if len(self.content[index]) > 0 and not self.content[index][0].isspace():
+
                 if row is not None:
                     body += row
                 row = nodes.row()
                 fields = self.get_fields_from_definition(self.content[index])
+
+                # If there is a group then fill first line of table with name of group, separators and description
+                if has_groups:
+                    if current_group != (fields[0]):
+                        row += nodes.entry('', nodes.paragraph('', nodes.Text("=====")))
+                        if has_group_desc:
+                            row += nodes.entry('', nodes.paragraph('', nodes.Text(fields[0])))
+                        else:
+                            row += nodes.entry('', nodes.paragraph('', nodes.Text(fields[0].split(" ", 1)[0])))
+
+                        row += nodes.entry('', nodes.paragraph('', nodes.Text("=====")))
+                        if has_group_desc:
+                            row += nodes.entry('', nodes.paragraph('', nodes.Text(fields[4])))
+                        else:
+                            row += nodes.entry('', nodes.paragraph('', nodes.Text("=====")))
+                        body += row
+                        current_group = (fields[0])
+                        pass
```

Comment:
Nedává smysl trochu jednodušší zápis?
```suggestion
                        separator = "====="
                        par = [
                            separator, 
                            fields[0] if has_group_desc else fields[0].split(" ", 1)[0],
                            fields[4] if has_group_desc else separator,
                            separator,
                        ]
                        for p in par:
                            row += nodes.entry('', nodes.paragraph('', nodes.Text(p)))
                        body += row
                        current_group = (fields[0])
```

Reply:
Jop, takhle je to daleko hezčí

Reply:
Díky tomu se mi podařilo odhalit překlep co tam byl :D